tests/codegen/intrinsics/transmute.rs

```diff
@@ -301,3 +327,65 @@ pub unsafe fn check_pair_to_array(x: (i64, u64)) -> [u8; 16] {
     // CHECK: store i64 %x.1, ptr %{{.+}}, align 1
     transmute(x)
 }
+
+// CHECK-LABEL: @check_heterogeneous_integer_pair(
+#[no_mangle]
+pub unsafe fn check_heterogeneous_integer_pair(x: (i32, bool)) -> (bool, u32) {
```

Comment:
What are the `store`s at the start of `check_heterogeneous_integer_pair`, `check_heterogeneous_float_pair` and `check_issue_110005`? From reading [LLVM LangRef](https://llvm.org/docs/LangRef.html#store-instruction) it looks like `store` is supposed to take a value and a pointer, but these only take values?...

I can't quite understand what is written here...

Reply:
The actual IR looks like this:
```llvm
define { float, double } @check_heterogeneous_float_pair(double noundef %x.0, float noundef %x.1) unnamed_addr #0 {
start:
  %0 = alloca { float, double }, align 8
  %1 = getelementptr inbounds { double, float }, ptr %0, i32 0, i32 0
  store double %x.0, ptr %1, align 8
  %2 = getelementptr inbounds { double, float }, ptr %0, i32 0, i32 1
  store float %x.1, ptr %2, align 8
  %3 = getelementptr inbounds { float, double }, ptr %0, i32 0, i32 0
  %4 = load float, ptr %3, align 8, !noundef !1
  %5 = getelementptr inbounds { float, double }, ptr %0, i32 0, i32 1
  %6 = load double, ptr %5, align 8, !noundef !1
  %7 = insertvalue { float, double } poison, float %4, 0
  %8 = insertvalue { float, double } %7, double %6, 1
  ret { float, double } %8
}
```
But the `CHECK` lines don't need to match the whole thing.

I wrote the test as a "what I really want to see is that it's loading/storing them", and didn't bother putting all the GEPing in the `CHECK`s since that would just clutter things up more.